### PR TITLE
Update PHPUnit and establish passing builds

### LINKED ISSUE #91: Cannot build or test locally
#### Issue Summary

I wanted to work on a fix for an issue reported here, but I am unable to run the existing tests locally or in CI, due to a test for files which do not exist in the repo. There is a test which checks for two Dockerfiles, which are not part of the repo, and the instructions to build and test locally don't include any info on creating these files. 

#### Steps to Reproduce

1. Clone the repo
2. composer install
3. run the tests as instructed in the docs

#### Technical details:

* smtpapi-php Version: master (latest commit: [729d8e8690e255559bc4d81d41546aff8d5fd987])
* PHP Version: 5.6 and 7.0

### PR DESCRIPTION
**Pull request by jmauerhan**
# Fixes # 
Fixes #91 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works

### Short description of what this PR does:
- Updates PHPUnit to the latest version which supports PHP 5.6
- The tests were *not* passing *or* running in CI. Before the broken file check test was added, the tests were not running. The build passed but if you look at the build log it says "No tests executed!" This fixes that. I ran 27 tests.
- Fixed one test that had the wrong reference to a file.

Passing tests on CI:
image

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/smtpapi-php?pullRequest=92) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/smtpapi-php?pullRequest=92) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/smtpapi-php?pullRequest=92) it.</sub>

**Comment by jmauerhan** [comment]
I'll figure out a better compatible version of PHP Unit and re-open.

**Comment by jmauerhan** [comment]
Well the problem still persists in CI. I was able to get it to start running the tests locally, and can successfully run the tests with the same command, but in CI it still reports no tests. I'll dig into it more on my own fork, sorry about that.

**Comment by jmauerhan** [comment]
Think I got it this time, got the tests to run on my Travis 😀 The biggest issue was the casing of the directory in the phpunit XML file.

File: composer.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
         "php": ">=5.6"
     },
     "require-dev": {
-        "phpunit/phpunit": "3.7.*"
+        "phpunit/phpunit": "5.7.*"
```
Comment (peter279k) [inline composer.json]: How about setting the `^5.7 || ^6.5` to support the `php7.2`?

Please refer the [official PHPUnit website](https://phpunit.de/).

File: composer.lock [hunk 2]
```diff
@@ -1,54 +1,364 @@
 {
```
Comment (peter279k) [inline composer.lock]: Is this file necessary? Once we update the dependencies, this file will be updated again.

I think it's fine to remove this locked file from `Git` version control and avoid caching the dependencies.

File: test/phpunit.xml.dist [hunk 3]
```diff
@@ -2,9 +2,8 @@
 <phpunit bootstrap="bootstrap.php">
   <testsuites>
     <testsuite name="General">
-      <file>./smtpapi.php</file>
-      <file>./smtpapi/Header.php</file>
-      <file>./RepoFilesExists.php</file>
+      <file>./Smtpapi.php</file>
+      <file>./Smtpapi/Header.php</file>
```
Comment (peter279k) [inline test/phpunit.xml.dist]: I don't think it's the proper way to define `tests` files with `file` tag.

How about using the `directory` tag to set the `tests` files?

File: composer.json (hunk 1 above)
Comment (jmauerhan) [reply composer.json]: Will do!

File: test/phpunit.xml.dist (hunk 3 above)
Comment (jmauerhan) [reply test/phpunit.xml.dist]: I'd agree, I was just working off what was already in the file :)

File: composer.lock (hunk 2 above)
Comment (jmauerhan) [reply composer.lock]: The lock file keeps the dependencies locked to the exact version they were at the time everything passed, so it's an extra precaution to ensure things don't break. This is an article that explains it better than I can: https://www.engineyard.com/blog/composer-its-all-about-the-lock-file

Here's the relevant section:

> Do You Know Where Your Lock File Is?
> A few weeks ago on Twitter, I noticed that the OpenCFP project doesn’t have a composer.lock file in it’s repository. “So what,” you might say, “just composer install and away you go. You’ll get the same dependencies, right?”
> 
> Wrong.
> 
> The point of the lock file is to record the exact versions that are installed so they can be re-installed. This means that if you have a version spec of 1.* and your co-worker runs composer update which installs 1.2.4, and then commits the composer.lock file, when you composer install, you will also get 1.2.4, even if 1.3.0 has been released. This ensures everybody working on the project has the same exact version.
> 
> Now, you might be thinking: with proper semantic versioning, 1.3.0 (or even 1.2.5) should be backwards compatible because it still conforms to the 1.* version spec, i.e. the major version number hasn’t changed. So what’s the big deal?
> 
> Well, unless you are performing code reviews on every release of every dependency, and their dependencies, and so on and so forth, it’s highly likely that some sort of bug or change will creep in eventually that will break your code. We’re only human, after all.
> 
> It’s also pretty common for a project to set a dependency version to rely on dev-master, meaning they pull the most recent changes each time. This means that if anything has been committed since the last time a composer install was done, then, without a lock file, you will get new third-party code being pulled down.
> 
> Again, this is a problem if you’re concerned about your code breaking. And it’s one of the reasons why it’s important to think about Composer as being centered around the composer.lock file. It is a safety device, and should be used like one.

File: test/phpunit.xml.dist (hunk 3 above)
Comment (jmauerhan) [reply test/phpunit.xml.dist]: Although if I make that change right now, I would have to add an explicit exception for the RepoFilesExists test that is broken. Maybe this is okay for now, and can be revisited once more tests are added?

**Comment by thinkingserious** [comment]
Hello @jmauerhan,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by thinkingserious** [comment]
Hello @jmauerhan,<br /><br/>Thanks again for the PR!<br /><br />                                         You have earned 6 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold or to simply continue the celebration.<br /><br />                                         Also, <strong>please be sure you have officially registered with us <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a> by November 1, 2018</strong> to qualify.<br /><br />                                         If you have any questions you can email us at <a href="mailto:dx+hacktoberfest2018@sendgrid.com">dx+hacktoberfest2018@sendgrid.com</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid
